File: backend/src/main/java/com/example/backend/Service/RideService.java
package com.example.backend.Service;

import com.example.backend.Entity.Ride;
import java.util.List;
import java.util.Map;

public interface RideService {
    Ride createRide(Long userId, String pickup, String drop, String vehicle, Double distance);
    List<Ride> getAvailableRides();
    Ride acceptRide(Long rideId, Long driverId);
    Ride startRide(Long rideId, String otp);
    Ride completeRide(Long rideId);
    List<Ride> getUserRides(Long userId);
    Ride getDriverActiveRide(Long driverId);
    Ride cancelRide(Long rideId);
}

File: frontend/src/services/rideService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/rides';

export const bookRide = async (userId, pickup, drop, vehicle, distance) => {
    const response = await axios.post(`${API_BASE_URL}/book`, {
        userId, pickup, drop, vehicle, distance
    });
    return response.data;
};

export const getAvailableRides = async () => {
    const response = await axios.get(`${API_BASE_URL}/available`);
    return response.data;
};

export const acceptRide = async (rideId, driverId) => {
    const response = await axios.post(`${API_BASE_URL}/accept`, {
        rideId, driverId
    });
    return response.data;
};

export const startRide = async (rideId, otp) => {
    const response = await axios.post(`${API_BASE_URL}/start`, {
        rideId, otp
    });
    return response.data;
};

export const completeRide = async (rideId) => {
    const response = await axios.post(`${API_BASE_URL}/complete`, {
        rideId
    });
    return response.data;
};

export const getUserRides = async (userId) => {
    const response = await axios.get(`${API_BASE_URL}/user/${userId}`);
    return response.data;
};
